File: rb/utility/prepare_path.rb
# PublibikeStations SDK utility: prepare_path
require_relative 'struct/voxgig_struct'
module PublibikeStationsUtilities
  PreparePath = ->(ctx) {
    point = ctx.point
    parts = []
    if point
      p = VoxgigStruct.getprop(point, "parts")
      parts = p if p.is_a?(Array)
    end
    VoxgigStruct.join(parts, "/", true)
  }
end

File: .sdk/tm/php/utility/PreparePath.php
<?php
declare(strict_types=1);

// PublibikeStations SDK utility: prepare_path

class PublibikeStationsPreparePath
{
    public static function call(PublibikeStationsContext $ctx): string
    {
        $point = $ctx->point;
        $parts = [];
        if ($point) {
            $p = \Voxgig\Struct\Struct::getprop($point, 'parts');
            if (is_array($p)) {
                $parts = $p;
            }
        }
        return \Voxgig\Struct\Struct::join($parts, '/', true);
    }
}
